### DISCUSSION TITLE
Cura UBL Start gcode settings - Check me please

### DISCUSSION THREAD

**Discussion by rourky80** [Jyers/Marlin#250] [Q&A] [answered]
I have the 15x15UBL firmware (preconfigured as it suited what I needed) and the mesh is completed and then stored via the screen. I wanted to check the start gcode to make the print use this mesh and then probe 3 for tilt (as UBL was designed) to get the best out of the BLTouch and UBL. What I have worked out is:

G28 ; Home all axes
M420 S1
G29 L1
G29 J
M500 
(then my extruder purge line code after this).

my take is M420 S1 enables bed leveling
G29 L1 loads the mesh stored in eeprom (assuming this is spot 1)
G29 J to probe for tilting and fix as required.
M500 to store the tilt --> is this required to use the G29 J tilting? or should i not do it as it would override my saved mesh?

first time using bed leveling, let me know if i'm good or way off track, thanks!

**Comment by Schnabulation** [comment, marked as answer]
Prefix: I know a lot about UBL but never used it just yet, so take my answer with a grain of salt.

I think to activate UBL you use `G29 A` instead of `M420 S1`. Also the `M500` os not needed I belive because you will tilt it before anyway before every print.

**Reply by rourky80** [reply]
Thanks for your help!
Yeah more thinking about it, I worked out that I wouldn't need the M500. 
I also found gcode from teachingtech for UBL that had:

G28 ; home all axes
G29 L1 ; Load the mesh stored in slot 1
G29 J ; Probe 3 points to tilt mesh
;M420 S1 ; restore ABL mesh    

So I think thats how I'll run it first to see. So at least I was on the right track.

**Reply by Jyers** [reply]
FYI G29 A and M420 S1 do essentially the same thing as far as I can tell
